pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -180,6 +180,25 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 		},
 	}
 
+	if options.DNSConfig != nil {
+		_, dnsPortStr, err := net.SplitHostPort(options.DNSConfig.BindAddress)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse DNS bind address %s: %v", options.DNSConfig.BindAddress, err)
+		}
+		dnsPort, err := strconv.Atoi(dnsPortStr)
+		if err != nil {
+			return nil, fmt.Errorf("invalid DNS port: %v", err)
+		}
+		m.ExtraServicePorts = append(m.ExtraServicePorts,
```

Comment:
can you confirm these get added if you start from an etcd which already has the service created without them?